include/boost/scope_exit.hpp

```diff
@@ -61,7 +61,7 @@
 #   define BOOST_SCOPE_EXIT_AUX_TPL_GCC_WORKAROUND_01 0
 #endif
 
-#if BOOST_MSVC
+#if BOOST_MSVC && (BOOST_MSVC <= 1900)
```

Comment:
Maybe use MSVC_TYPEOF_HACK instead to make sure that future changes to config or typeof don't break anything?

Reply:
Wouldn't < 1901 be better (assuming this was updated in 1901)?